src/Internal/DependencyManagement/ServiceProvider/PaymentsServiceProvider.php

```diff
@@ -41,6 +44,7 @@ class PaymentsServiceProvider extends AbstractServiceProvider {
 		Router::class,
 		StateFactory::class,
 		InitialState::class,
+		ProcessedState::class,
```

Comment:
Just like this declaration, we need to declare that the provider provides `AuthenticationRequiredState`. Right now, everything works, because the `register` method is called for another state. At some point in the future, if someone requests the `AuthenticationRequiredState` directly, this could lead to a fatal error.